wd_helper.au3

```diff
@@ -672,6 +679,158 @@ Func _WD_FrameLeave($sSession)
+;                  $sFrameAttributes    - all <iframe ....> atributes
+; Return values .: Success - array or string
+;                  Failure - "" (empty string) and sets @error to one of the following values:
+;                  - $_WD_ERROR_Exception
+; Author ........: mLipok
+; Modified ......: Danp2
+; Remarks .......:
+; Related .......: _WD_FrameList, _WD_GetFrameCount, _WD_FrameEnter, _WD_FrameLeave
+; Link ..........:
+; Example .......: No
+; ===============================================================================================================================
+Func __WD_FrameList_Internal($sSession, $sLevel, $sFrameAttributes)
+	Local Const $sFuncName = "__WD_FrameList_Internal"
+	Local Const $sParameters = 'Parameters:    Level=' & $sLevel ; intentionally $sFrameAttributes is not listed here to not put too many data into the log
+	Local $iErr = $_WD_ERROR_Success
+	Local $vResult = '', $s_URL = '', $sMessage = ''
+
+	_WD_FrameEnter($sSession, $sLevel)
+	$iErr = @error
+
+	If @error Then
+		$sMessage = 'Error occured on "' & $sLevel & '" level when trying to entering frame'
+	Else
+		Local $sCurrentBody_ElementID = _WD_ExecuteScript($sSession, "return window.document.body;", Default, Default, $_WD_JSON_Element)
+		$iErr = @error
+		If @error Then
+			$sMessage = 'Error occured on "' & $sLevel & '" level when checking "document.body" ElementID'
+		Else
+			$s_URL = _WD_ExecuteScript($sSession, "return window.location.href", Default, Default, $_WD_JSON_Value)
+			$iErr = @error
+			If @error Then
+				$sMessage = 'Error occured on "' & $sLevel & '" level when checking URL'
+			Else
+				$vResult = $sLevel & '|' & $sLevel & '|' & $sFrameAttributes & '|' & $s_URL & '|' & $sCurrentBody_ElementID & @CRLF
+			EndIf
+		EndIf
+	EndIf
+
+	If Not @error Then
+		Local $iFrameCount = _WD_GetFrameCount($sSession)
+		$iErr = @error
+		If $iErr Then
+			$sMessage = 'Error occured on "' & $sLevel & '" level when trying to check frames count'
+		Else
+			For $iFrame = 0 To $iFrameCount - 1
+				$sFrameAttributes = _WD_ExecuteScript($sSession, "return document.querySelectorAll('iframe')[" & $iFrame & "].outerHTML;", Default, Default, $_WD_JSON_Value)
+				$iErr = @error
+				If @error Then
+					$sMessage = 'Error occured on "' & $sLevel & '" level when trying to check atributes child frames #' & $iFrame
+				Else
+					$sFrameAttributes = StringRegExpReplace($sFrameAttributes, '\R', '')
+					$vResult &= __WD_FrameList_Internal($sSession, $sLevel & '/' & $iFrame, $sFrameAttributes)
+					$iErr = @error
+					If Not @error Then
+						_WD_FrameLeave($sSession)
+						$iErr = @error
+						If @error Then
+							$sMessage = 'Error occured on "' & $sLevel & '" level when trying to leave frames #' & $iFrame
+						EndIf
+					EndIf
+				EndIf
+				If @error Then ExitLoop
+			Next
+		EndIf
+	EndIf
+	If $iErr Then $iErr = $_WD_ERROR_Exception
+
+	$sMessage = ($sMessage And $_WD_DEBUG = $_WD_DEBUG_Full) ? (' Information: ' & $sMessage) : ("")
```

Comment:
I still propose here:
$sMessage = ($sMessage And $_WD_DEBUG > $_WD_DEBUG_Error) ? (' Information: ' & $sMessage) : ("")

Reply:
That should be ok.

Reply:
You could finish this up at the same time. 😉

Reply:
Take a look on: [$sMessage on > $_WD_DEBUG_Error](https://github.com/Danp2/au3WebDriver/pull/362/commits/acb79955b41f168029ed70b5065a8acbb759ee6a)